software/util/buildsite.py

```diff
@@ -121,36 +128,42 @@ def createMissingDir(dir):
     if not os.path.exists(dir):
         os.makedirs(dir)
 
+HANDLER_TEMPLATE = 'handlers-template.yaml'
+HANDLER_FILE = 'handlers.yaml'
+
 def initdir():
-    print("Building site in '%s' directory" % OUTPUTDIR)
+    print('Building site in "%s" directory' % OUTPUTDIR)
     createMissingDir(OUTPUTDIR)
     clear()
-    createMissingDir(OUTPUTDIR + "/docs")
-    createMissingDir(OUTPUTDIR + "/docs/contributors")
-    createMissingDir(OUTPUTDIR + "/empty") #For apppengine 404 handler
-    createMissingDir(OUTPUTDIR + "/releases/%s" % getVersion())
+    createMissingDir(os.path.join(OUTPUTDIR, 'docs'))
+    createMissingDir(os.path.join(OUTPUTDIR, 'docs/contributors'))
+    createMissingDir(os.path.join(OUTPUTDIR, 'empty')) #For apppengine 404 handler
+    createMissingDir(os.path.join(OUTPUTDIR, 'releases', getVersion()))
 
-    gdir = OUTPUTDIR + "/gcloud"
+    gdir = os.path.join(OUTPUTDIR, 'gcloud')
     createMissingDir(gdir)
 
-    print("\nCopying docs static files")
-    cmd = "./software/util/copystaticdocsplusinsert.py"
-    os.system(cmd)
-    print("Done")
-
-    print("\nPreparing GCloud files")
-    cmd = "cp software/gcloud/*.yaml %s" % gdir
-    os.system(cmd)
-    print("Files copied")
+    print('\nCopying docs static files')
+    cmd = ['./software/util/copystaticdocsplusinsert.py']
+    subprocess.check_call(cmd)
+    print('Done')
+
+    print('\nPreparing GCloud files')
+    gcloud_files = glob.glob('software/gcloud/*.yaml')
+    for path in gcloud_files:
+      shutil.copy(path, gdir)
+    print('Done: copied %d files' % len(gcloud_files))
+    print('\nCreating %s from %s for version: %s' % (HANDLER_FILE, HANDLER_TEMPLATE, getVersion()))
+    with open(os.path.join(gdir, HANDLER_TEMPLATE)) as template_file:
+      with open(os.path.join(gdir, HANDLER_FILE), mode='w') as yaml_file:
+        template_data = template_file.read()
```

Comment:
should this line be outside the loop? Can you read the same data multiple times from an opened file?